.githooks/pre-commit

```diff
@@ -22,3 +22,6 @@ for FILE in ${STAGED_EVG_FILES}
 do
     evergreen validate "${FILE}"
 done
+
+make gen-docs > /dev/null
+git add docs
```

Comment:
[nit] I think it makes sense to only run this inside `if [[ -n "${STAGED_GO_FILES}" ]]; then`